File: sample/src/main/java/com/like/wxpay/sample/MainActivity.java
package com.like.wxpay.sample;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import com.like.wxpay.WXPayUtils;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        WXPayUtils.getInstance(this).init("wxb4ba3c02aa476ea1");
        WXPayUtils.getInstance(this).pay("{\"appid\":\"wxb4ba3c02aa476ea1\",\"partnerid\":\"1900006771\",\"package\":\"Sign=WXPay\",\"noncestr\":\"9ed4e2f1a00597bcd5d6a2c88094760d\",\"timestamp\":1521008462,\"prepayid\":\"wx201803141421023e376ecbde0653253070\",\"sign\":\"9E753150CC1FED6783CF3FBD10DC89A4\"}");
    }
}

File: sample/src/main/java/com/like/wxpay/sample/MainActivity.kt
package com.like.wxpay.sample

import android.os.Bundle
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.like.floweventbus.FlowEventBus
import com.like.floweventbus_annotations.BusObserver
import com.like.wxpay.WxPayUtils

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        WxPayUtils.init(this, "wxb4ba3c02aa476ea1")
        FlowEventBus.register(this)
    }

    fun pay(view: View) {
        WxPayUtils.pay(
            this,
            "{\"appid\":\"wxb4ba3c02aa476ea1\",\"partnerid\":\"1900006771\",\"package\":\"Sign=WXPay\",\"noncestr\":\"ffd8363d5149de96acfb21f861b6f2d7\",\"timestamp\":1546589467,\"prepayid\":\"wx04161107414587fcf981bea61635925982\",\"sign\":\"3618C1D08BC50D94F2952C3C3A7F7B3B\"}"
        )
    }

    @BusObserver([WxPayUtils.TAG_PAY_SUCCESS])
    fun onPaySuccess() {
        Toast.makeText(this, "微信支付成功", Toast.LENGTH_SHORT).show()
    }

    @BusObserver([WxPayUtils.TAG_PAY_FAILURE])
    fun onPayFailure() {
        Toast.makeText(this, "微信支付失败", Toast.LENGTH_SHORT).show()
    }
}